src/uu/stty/src/stty.rs

```diff
@@ -264,12 +264,36 @@ fn stty(opts: &Options) -> UResult<()> {
         ));
     }
 
+    // Fetch termios for the target device once and reuse it downstream
+    let base_termios = tcgetattr(opts.file.as_fd())?;
```

Comment:
IMHO this can be still just an Option set to None here, and you set it when needed, so before parsing, or - if unset - before applying the arguments.